### ISSUE TITLE
Should `draw_image` close the file descriptor?

### ISSUE THREAD

**Issue by aried3r** [hidakatsuya/prawn-emoji#29]
Hey!

I was wondering if `draw_image` shouldn't maybe close the file descriptor of `image`. In our usage of prawn the tend to close the file descriptor when printing images, but I'm not sure if that would somehow affect prawn-emoji in any way or if we should trust the GC with that?

https://github.com/hidakatsuya/prawn-emoji/blob/26f83a1c31c70642291d3b2d4a2ddff6379dbe57/lib/prawn/emoji/drawer.rb#L45-L54

After rendering a PDF I get the following output from `ObjectSpace`:

```Ruby
require 'prawn/emoji'

Prawn::Document.generate 'sushi.pdf' do
  font '/Users/kakazza/Library/Fonts/DejaVuSans.ttf'
  text '🐟 / 🔪 + 🍚 / 🍾 = 🍣'
end

ObjectSpace.each_object(File) do |f|
  puts "%s: %d" % [f.path, f.fileno] unless f.closed?
end

DejaVuSans.ttf: 26
prawn-emoji-3.1.0/emoji/images/1f35a.png: 29
prawn-emoji-3.1.0/emoji/images/1f37e.png: 30
prawn-emoji-3.1.0/emoji/images/1f363.png: 31
prawn-emoji-3.1.0/emoji/images/1f41f.png: 27
prawn-emoji-3.1.0/emoji/images/1f52a.png: 28
```

**Comment by hidakatsuya** [comment]
Thank you for your reporting. It's very interesting for me. I'll look into it later.

**Comment by hidakatsuya** [comment]
prawn-emoji just passes the path of image file to `Prawn#image` method but not open image file.
https://github.com/hidakatsuya/prawn-emoji/blob/26f83a1c31c70642291d3b2d4a2ddff6379dbe57/lib/prawn/emoji/drawer.rb#L53

Image files are opened by prawn. Please see the issue below for details:
https://github.com/prawnpdf/prawn/issues/975

This issue has been already fixed by https://github.com/prawnpdf/prawn/pull/1090, but not released yet. (I strongly hope it will be released)

> but I'm not sure if that would somehow affect prawn-emoji in any way or if we should trust the GC with that?

In the code below, the GC removes those file descriptors. GC seems to work as expected.

```ruby
require 'prawn'

def inspect_opened_files
  ObjectSpace.each_object(File) do |f|
    puts "%s: %d" % [f.path, f.fileno] unless f.closed?
  end
end

Prawn::Document.generate 'test.pdf' do
  image 'ruby.png'
end

puts 'before GC:'
inspect_opened_files

ObjectSpace.garbage_collect

puts 'after GC:'
inspect_opened_files
```

```
before GC:
ruby.png: 5
after GC:

```

However, I'm thinking of fixing the prawn-emoji to close the file descriptor properly.

**Comment by hidakatsuya** [comment]
Fixed by #31 and automatically closed. Please feel free to reopen if you'd like.
